server/api.js

```diff
@@ -316,50 +322,56 @@ router.post("/bid/:bidId/status", async (req, res) => {
 
 		await client.query("BEGIN");
 
-		const tenderResult = await client.query(
-			"SELECT tender_id FROM bid WHERE bid_id = $1;",
+		const bidResult = await client.query(
+			"SELECT tender_id, bidder_id FROM bid WHERE bid_id = $1;",
 			[bidId]
 		);
 
-		if (tenderResult.rowCount === 0) {
+		if (bidResult.rowCount === 0) {
 			await client.query("ROLLBACK");
 			return res.status(404).send({ code: "BID_NOT_FOUND" });
 		}
 
-		const tenderId = parseInt(tenderResult.rows[0].tender_id);
-
-		if (status === "Awarded") {
-			const rejectStatus = "Rejected";
-			const awardBidResult = await client.query(
-				"UPDATE bid SET status = $1 WHERE tender_id = $2 AND bid_id = $3;",
-				[status, tenderId, bidId]
-			);
-
-			const rejectOtherBidsResult = await client.query(
-				"UPDATE bid SET status = $1 WHERE tender_id = $2 AND bid_id != $3;",
-				[rejectStatus, tenderId, bidId]
-			);
-
-			if (awardBidResult.rowCount > 0 || rejectOtherBidsResult.rowCount > 0) {
-				await client.query("COMMIT");
-				return res.status(200).send({ code: "SUCCESS" });
-			} else {
-				await client.query("ROLLBACK");
-				return res.status(500).send({ code: "SERVER_ERROR" });
-			}
+		const bid = bidResult.rows[0];
+		const tenderId = bid.tender_id;
+		const bidderId = bid.bidder_id;
+
+		const tenderResult = await client.query(
+			"SELECT buyer_id FROM tender WHERE id = $1;",
+			[tenderId]
+		);
+
+		if (tenderResult.rowCount === 0) {
+			await client.query("ROLLBACK");
+			return res.status(404).send({ code: "TENDER_NOT_FOUND" });
+		}
+
+		const buyerId = tenderResult.rows[0].buyer_id;
+
+		if (
+			(status === "Awarded" || status === "Rejected") &&
+			user.id !== buyerId
+		) {
+			await client.query("ROLLBACK");
+			return res.status(403).send({ code: "FORBIDDEN" });
```

Comment:
When a bid is awarded, we need to reject all the other bids on the same tender. Why did you remove that code?